### ISSUE TITLE
Switch to Eigen and remove dependence on TooN

### ISSUE THREAD

**Issue by stevenwaslander** [wavelab/mcptam#16]

**Comment by SabaAflaki** [comment]
Should I use Eigen in EKF code?

**Comment by stevenwaslander** [comment]
Yes, that would be good, assuming it has everything we need? The key would be to properly handle rotation matrices and homogeneous transforms.  We want to work in SE(3).  Is that possible? TooN was very nice for this, and in the MCPTAM code, it worked well with g2o as a result.  I noticed that GTSAM also does SE(3) well, so if Eigen can do that for us, then we are in good shape.  If not, you could suggest something else.

**Comment by StanJBrown** [comment]
Eigen does not support SE3 calculations out of the box. However there are repos that people have made such as this one that add the support for SE3. https://github.com/strasdat/Sophus

**Comment by stevenwaslander** [comment]
That could work, code hasn't been touched in three years, so we could pull a copy in as part of mcptam and freeze it.  Only issue is that it is not very standard, I don't think.  It would remove TooN as a dependency, but require documentation of this SE(3) class.  No license issue either.

**Comment by StanJBrown** [comment]
Oh I was meaning more of we probably would have to write our own code to do it. We could use this as an example of how to do with Eigen, but ultimately write our own stuff so we know how it works.

**Comment by SabaAflaki** [comment]
This looks good too:
https://www.sofa-framework.org/api/Compliant/struct_s_e3.html
